pkg/cloudprovider/providers/azure/azure_test.go

```diff
@@ -354,14 +354,14 @@ func TestReconcileSecurityGroupRemoveServiceRemovesPort(t *testing.T) {
 
 func TestReconcileSecurityWithSourceRanges(t *testing.T) {
 	az := getTestCloud()
-	svc := getTestService("servicea", v1.ProtocolTCP, 80, 443)
+	svc := getTestService("servicea", v1.ProtocolTCP, "192.168.0.0", 80, 443)
 	svc.Spec.LoadBalancerSourceRanges = []string{
 		"192.168.0.0/24",
 		"10.0.0.0/32",
 	}
 
 	sg := getTestSecurityGroup(svc)
-	sg, _, err := az.reconcileSecurityGroup(sg, testClusterName, &svc, to.StringPtr("192.168.0.0"), true)
+	sg, _, err := az.reconcileSecurityGroup(sg, testClusterName, &svc, to.StringPtr(svc.Spec.LoadBalancerIP), true)
```

Comment:
Can we add a test which explicitly ensures that we reconcile when switching from an empty load balancer IP ("") to a valid one? This is the exact case we are hoping to fix with this PR, going from Kube pre-1.8 to 1.8.

Reply:
You mean `DestinationAddressPrefix` of `Any` to valid IP that corresponds to Service loadbalancer IP? I will add a test that tests it.

Reply:
Might also refer to the case where the service spec does not specify a LoadBalancerIP so one has been assigned by Azure.